### DISCUSSION TITLE
timestamp type error

### DISCUSSION THREAD

**Discussion by Alkarex** [alexandrainst/node-red-contrib-postgresql#43] [Q&A] [answered]
Hello,
every things ok  but....
I'd like to insert timestamp in my database, this the last value of my query, but I don't know why the format is changed as we can see in this capture:
![image](https://user-images.githubusercontent.com/85061112/230479333-f1c4aa02-9f4e-418b-af0c-ca184e19430a.png)

Why the query convert the time in GMT ?

_Originally posted by @FlorianBfr in https://github.com/alexandrainst/node-red-contrib-postgresql/discussions/41#discussioncomment-5547145_

**Comment by Alkarex** [comment, marked as answer]
This looks like because you are likely mislead by your debug output. You are sending a JavaScript DateTime object while you should be sending a string (in ISO format).

**Comment by FlorianBfr** [comment]
Just before I have a function where I convert the orignal timestamp in date :
     var d = new Date (msg.payload.tst*1000);
     newpayload.event_datetime = d;
     msg.payload = newpayload;
     return msg;

**Reply by Alkarex** [reply]
This is exactly what is wrong. You need to convert your date object to a string

**Reply by FlorianBfr** [reply]
When I convert the date in string the trouble is the same
![image](https://user-images.githubusercontent.com/85061112/230600749-dd699f9e-201d-455b-a115-a9103b605083.png)

**Reply by Alkarex** [reply]
As string, but in the correct format!
You need an ISO format

**Reply by FlorianBfr** [reply]
Ok I will try, thank

**Reply by FlorianBfr** [reply]
Yess ! You are very good !
Tank you very much. The good sentence is : 
var d = new Date(msg.payload.tst*1000);
newpayload.event_datetime = d.toISOString();
